bindings/trycp_client/src/lib.rs

```diff
@@ -0,0 +1,888 @@
+                .await?;
+
+            check_empty_response(response)
+        }
+
+        /// Hook up an app interface.
+        ///
+        /// Provide a `token` that has been issued using the admin interface and the `port` to
+        /// connect to.
+        #[wind_tunnel_instrument(prefix = "trycp_")]
+        pub async fn connect_app_interface(
+            &self,
+            token: Vec<u8>,
+            port: u16,
+            timeout: Option<Duration>,
+        ) -> io::Result<()> {
+            let response = self
+                .trycp_client
+                .request(
+                    Request::ConnectAppInterface { token, port },
+                    timeout.unwrap_or(self.timeout),
+                )
+                .await?;
+
+            check_empty_response(response)
+        }
+
+        /// Disconnect an app interface.
+        ///
+        /// Provide the `port` to identify the app interface to disconnect.
+        #[wind_tunnel_instrument(prefix = "trycp_")]
+        pub async fn disconnect_app_interface(
+            &self,
+            port: u16,
+            timeout: Option<Duration>,
+        ) -> io::Result<()> {
+            let response = self
+                .trycp_client
+                .request(
+                    Request::DisconnectAppInterface { port },
+                    timeout.unwrap_or(self.timeout),
+                )
+                .await?;
+
+            check_empty_response(response)
+        }
+    }
+}
+
+mod admin_impl {
+    use holo_hash::{AgentPubKey, DnaHash};
+    use holochain_client::{
+        AuthorizeSigningCredentialsPayload, EnableAppResponse, SigningCredentials,
+    };
+    use holochain_conductor_api::{
+        AdminRequest, AdminResponse, AppAuthenticationTokenIssued, AppInfo, AppInterfaceInfo,
+        AppStatusFilter, IssueAppAuthenticationTokenPayload, StorageInfo,
+    };
+    use holochain_types::app::{DeleteCloneCellPayload, InstallAppPayload, InstalledAppId};
+    use holochain_types::prelude::{GrantedFunctions, Record};
+    use holochain_types::websocket::AllowedOrigins;
+    use holochain_zome_types::capability::GrantZomeCallCapabilityPayload;
+    use holochain_zome_types::cell::CellId;
+    use holochain_zome_types::dna_def::DnaDef;
+
+    use super::*;
+
+    impl TryCPClientInstrumented {
+        #[wind_tunnel_instrument(prefix = "trycp_admin_")]
+        pub async fn generate_agent_pub_key(
+            &self,
+            id: String,
+            timeout: Option<Duration>,
+        ) -> io::Result<AgentPubKey> {
+            let response = self
+                .call_admin(id, AdminRequest::GenerateAgentPubKey, timeout)
+                .await?;
+
+            match response {
+                AdminResponse::AgentPubKeyGenerated(agent_pub_key) => Ok(agent_pub_key),
+                _ => Err(io::Error::new(
+                    io::ErrorKind::Other,
+                    "Unexpected admin response",
+                )),
```

Comment:
```suggestion
                _ => Err(io::Error::other("Unexpected admin response")),
```

[other](https://doc.rust-lang.org/std/io/struct.Error.html#method.other) has been stabilized in 1.74

Reply:
Much tidier, blame copilot! ;) Changed everywhere